python/cudf/cudf/_lib/CMakeLists.txt

```diff
@@ -63,6 +63,7 @@ rapids_cython_create_modules(
 )
 
 target_link_libraries(strings_udf PUBLIC cudf_strings_udf)
+target_include_directories(interop PUBLIC "$<BUILD_INTERFACE:${DLPACK_INCLUDE_DIR}>")
```

Comment:
This was in the diff before I got to this PR, and I convinced myself that I understood why. Please do correct my understanding (and @msarahan might remember as well).

I *think* this was required before, but `cudf` and `pylibcudf` were getting these headers by downloading them via CPM:

https://github.com/rapidsai/cudf/blob/e4e867aace96b80fccf030cc02a11f89cbb9c05f/python/cudf/CMakeLists.txt#L58-L60

https://github.com/rapidsai/cudf/blob/e4e867aace96b80fccf030cc02a11f89cbb9c05f/python/pylibcudf/CMakeLists.txt#L60

That's behavior that's preserved here, for example by downloading those same headers in `python/pylibcudf/CMakeLists.txt` and `python/cudf/CMakeLists.txt`, like this:

```cmake
# an installed version of libcudf doesn't provide the dlpack headers so we need to download dlpack
# for the interop.pyx
include(rapids-cpm)
rapids_cpm_init()
include(../../cpp/cmake/thirdparty/get_dlpack.cmake)
```

It's that call that sets the `DLPACK_INCLUDE_DIR`, I think.

But that include directory wasn't getting threaded through to the Cython targets with the changes in this PR. I'm not certain why. This `target_include_directories()` is here to ensure they can link with `libcudf` correctly, but not need yet another CPM download of `dlpack`.

I can say for sure that when I tried removing these, compiling `libcudf` succeeded but compiling `pylibcudf` failed, like this: https://github.com/rapidsai/cudf/pull/15483#issuecomment-2059520151

Reply:
You are correct about where `DLPACK_INCLUDE_DIR` is coming from and the fact that we were CPM downloading before. I am not sure why the `target_include_directories` is needed now when it wasn't before, but I'm OK punting on this question for now to get this PR merged in the hopes that we can improve our handling of dlpack headers during the build later. They require a bit more work anyway.

Reply:
OK I see what is happening here. The dlpack headers are only part of the public interface of the libcudf builds, not installations, and since this PR moves the cudf/pylibcudf packages from using a build tree to an install tree of libcudf these directories are no longer visible. These commands are correct.